opm/simulators/wells/BlackoilWellModel.hpp

```diff
@@ -184,7 +184,20 @@ namespace Opm {
             void initFromRestartFile(const RestartValue& restartValues);
 
             Opm::data::Wells wellData() const
-            { return well_state_.report(phase_usage_, Opm::UgGridHelpers::globalCell(grid())); }
+            {
+                auto wsrpt = well_state_.report(phase_usage_, Opm::UgGridHelpers::globalCell(grid()));
+
+                for (const auto& well : this->wells_ecl_) {
+                    auto xwPos = wsrpt.find(well.name());
+                    if (xwPos == wsrpt.end()) { // No well results.  Unexpected.
+                        continue;
```

Comment:
Better to throw? Or can this happen in some corner cases. I thought wellMap and wells_ecl always had the same wells.

Reply:
> I thought `wellMap` and `wells_ecl` always had the same wells.

I think they do, and I'd be really surprised if this condition actually happened.  As for throwing an exception here, that's a little tenuous.  This function runs in parallel (I think) so exceptions are delicate.  I'd be happy to add an `assert` here instead although that too is troublesome in a parallel context.